ntoskrnl/ex/sysinfo.c

```diff
@@ -1362,7 +1356,9 @@ QSI_DEF(SystemHandleInformation)
                     /* Check user's buffer size */
                     if (*ReqSize > Size)
                     {
+                        ExUnlockHandleTableEntry(HandleTable, HandleTableEntry);
                         Status = STATUS_INFO_LENGTH_MISMATCH;
+                        goto Abort;
```

Comment:
Please write an apitest that shows whether this is the behaviour happening on Windows, namely, whether `HandleInformation->NumberOfHandles` will return the number of handles stored in the user buffer, or will return the number of handles in the system, regardless of the size of the user buffer.

Reply:
Adding that shortcut was wrong.
I'm working on extending ntdll:info winetest, separately.